bundles/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/internal/discovery/TiVoDiscoveryParticipant.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.tivo.internal.discovery;
+
+import static org.openhab.binding.tivo.TiVoBindingConstants.*;
+
+import java.net.InetAddress;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The Class TiVoDiscoveryParticipant.
+ * *
+ *
+ * @author Jayson Kubilis (DigitalBytes) - Initial contribution
+ * @author Andrew Black (AndyXMB) - minor updates.
+ * @author Michael Lobstein - Updated for OH3
+ */
+@NonNullByDefault
+@Component(immediate = true, configurationPid = "discovery.tivo")
+public class TiVoDiscoveryParticipant implements MDNSDiscoveryParticipant {
+    private final Logger logger = LoggerFactory.getLogger(TiVoDiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(THING_TYPE_TIVO);
+    }
+
+    @Override
+    public String getServiceType() {
+        return "_tivo-remote._tcp.local.";
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        DiscoveryResult result = null;
+
+        ThingUID uid = getThingUID(service);
+        if (uid != null) {
+            Map<String, Object> properties = new HashMap<>(2);
+            // remove the domain from the name
+            InetAddress ip = getIpAddress(service);
+            if (ip == null) {
+                return null;
+            }
+            String inetAddress = ip.toString().substring(1); // trim leading slash
+            String label = service.getName();
+            int port = service.getPort();
+
+            properties.put(CONFIG_HOST, inetAddress);
+            properties.put(CONFIG_PORT, port);
+
+            result = DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel("Tivo: " + label)
+                    .withProperty(CONFIG_NAME, label).withRepresentationProperty(CONFIG_NAME).build();
```

Comment:
The representation property should be something more unique like the ip address.